### PR TITLE
ADLS Gen2 API Implementation

### PR DESCRIPTION
**Pull request by xiafu-msft**
* ADLS Gen2 API Implementation
(the namespace will be changed to azure.storage.filedatalake later)

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-file-datalake/tests/test_directory.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+        return directory_client
+
+    def _create_file_system(self):
+        return self.dsc.create_file_system(self._get_file_system_reference())
+
+
+    # --Helpers-----------------------------------------------------------------
+
+    @record
+    def test_create_directory(self):
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file-datalake/tests/test_directory.py]: Please log an issue to eventually migrate these tests to the new framework

Comment (xiafu-msft) [reply sdk/storage/azure-storage-file-datalake/tests/test_directory.py]: Sure! Created here ^_^ https://msazure.visualstudio.com/One/_workitems/edit/5703256

File: sdk/storage/azure-storage-file-datalake/setup.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    packages=find_packages(exclude=[
+        # Exclude packages that will be covered by PEP420 or nspkg
+        'azure',
+        'azure.storage',
+        'tests',
+    ]),
+    install_requires=[
+        "azure-core<2.0.0,>=1.0.0",
+        "msrest>=0.6.10",
+        "azure-storage-blob>=12.0.0"
```
Comment (zezha-msft) [inline sdk/storage/azure-storage-file-datalake/setup.py]: >> [](start = 27, length = 1)

change to ~=

File: sdk/storage/azure-storage-file-datalake/README.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
+# Azure DataLake service client library for Python
+Overview
+
+This preview package for Python includes ADLS Gen2 specific API support made available in Storage SDK. This includes:
+1. New directory level operations (Create, Rename/Move, Delete) for both hierarchical namespace enabled (HNS) storage accounts and HNS disabled storage accounts. For HNS enabled accounts, the rename/move operations are atomic.
```
Comment (zezha-msft) [inline sdk/storage/azure-storage-file-datalake/README.md]: >disabled [](start = 135, length = 8)

we support enabled only

File: sdk/storage/azure-storage-file-datalake/README.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+#### Clients
+
+The DataLake Storage SDK provides four different clients to interact with the DataLake Service:
+1. **DataLakeServiceClient** - this client interacts with the DataLake Service at the account level. 
+    It provides operations to retrieve and configure the account properties
+    as well as list, create, and delete file systems within the account.
+    For operations relating to a specific file system, directory or file, clients for those entities
+    can also be retrieved using the `get_file_client`, `get_directory_client` or `get_file_system_client` functions.
+2. **FileSystemClient** - this client represents interaction with a specific
+    file system, even if that file system need not exist yet. It provides operations to create, delete, or
```
Comment (zezha-msft) [inline sdk/storage/azure-storage-file-datalake/README.md]: >need [](start = 42, length = 4)

does

File: sdk/storage/azure-storage-file-datalake/README.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+* [`datalake_samples_access_control.py`](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/storage/azure-storage-file-datalake/samples/datalake_samples_access_control.py) - Examples for common DataLake Storage tasks:
+    * Set up a file system
+    * Create a directory
+    * Set/Get access control for the directory
+    * Create files under the directory
+    * Set/Get access control for each file
+    * Delete file system
+
+* [`datalake_samples_upload_download.py`](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/storage/azure-storage-file-datalake/samples/datalake_samples_upload_download.py) - Examples for authenticating and creating the client:
```
Comment (zezha-msft) [inline sdk/storage/azure-storage-file-datalake/README.md]: > Examples for authenticating and creating the client [](start = 186, length = 52)

Examples for common data operations

**Pull request merged**